### ISSUE TITLE
Typescript 2.7?

### ISSUE THREAD

**Issue by PavelPZ** [CompuIves/codesandbox-client#636]
Hi, I am experimenting with your great tool (thanks a lot for it).

**Are you using last Typescript version (2.7)? It allows this typescript definition file (e.g. ```index.d.ts```)**:

```
import * as React from "react";
import { render } from "react-dom";

const enum Consts {
  Drawer = 'AnyPropName'
}
interface Shapes {
  [Consts.Drawer]: string //<=== Error here
  OtherPropName: string
}
let X: Shapes = { 
  AnyPropName: '', //<=== Error here
  OtherPropNamex:''}

X = null

render(<div>Hallo world</div>, document.getElementById("root"));
```
CodeSandbox shows ```A computed property name in an interface must directly refer to a built-in symbol.``` error.

See
[![Edit k0pxwv2k5o](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/k0pxwv2k5o)

**Comment by CompuIves** [comment]
Hey @PavelPZ, our bundler uses TypeScript 2.7, but it seems like that Monaco is using an older version still. I've read that Monaco will get an update to the latest version soon, so we'll update as soon as that's in!

**Comment by PavelPZ** [comment]
Hi @CompuIves, thanks for info.

I thing that problem is more serious. It looks like CodeSandbox compiles project *file by file* without preserving any global contexts. See 
[![Edit 0m3plorznl](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/0m3plorznl)
and error in line 9 in index.tsx. 

Is any chance to solve it (at least in the future)? I am working on framework which enables creation of visual components for both react and react-native. E.g. I need to use typing (just typing, not code) from one platform (react-native) in the second platform (web) **without importing react-native package**. 
I cannot find any solotion which is working in CodeSandbox. 

I tried:
- shared *files.d.ts*, it needs to use *include* in tsconfig.json (explained in [637](https://github.com/CompuIves/codesandbox-client/issues/637))
- ```export as namespace```
- ```declare global {```

**Comment by PavelPZ** [comment]
... a little progress, see 
[![Edit mor043329x](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/mor043329x)

```export as namespace``` does not show error, but 
- typing is loosed
- cannot share ```const enum {}```

**Comment by CompuIves** [comment]
Ah, this is really good info. We indeed transpile file by file, and want to let the editor handle all error messages about types. This creates a clear separation of concerns. I think many problems will be solved when we move to Monaco v11 (#647), I need to do some more testing on there to make sure everything works.

**Comment by SaraVieira** [comment]
This appears to be fixed in the new monaco and vscode :D
